miri-script/src/commands.rs

```diff
@@ -499,9 +499,12 @@ impl Command {
         let mut early_flags = Vec::<OsString>::new();
 
         // Add target, edition to flags.
-        if let Some(target) = &target {
-            early_flags.push("--target".into());
-            early_flags.push(target.into());
+        // For `--dep` we need to set the env var instead of passing the flag.
+        if !dep {
+            if let Some(target) = &target {
+                early_flags.push("--target".into());
+                early_flags.push(target.into());
+            }
         }
```

Comment:
hmm now it's not setting the target anymore, before it was setting it twice. not sure when I'll get around to debugging this, probably not this week